addons/hr_holidays/models/hr_leave.py

```diff
@@ -1207,6 +1151,83 @@ def action_approve(self, check_state=True):
     def _get_leaves_on_public_holiday(self):
         return self.filtered(lambda l: l.employee_id and not l.number_of_days)
 
+    def _split_leaves(self, date_from, date_to):
+        """
+        Split leaves on the given full-day interval.
+        Some important notes:
+            - date_to is not included in the interval, so to split on an
+              'empty' interval (the boundary between two days), date_from
+              and date_to should be the same.
+            - This method only splits leaves on full-day intervals. Logic to
+              split leaves on partial days or hours is not straightforward as
+              you have to take into account working hours and timezones. It's
+              also not clear that we would want to handle this automatically.
+            - The method will therefore also only work on leaves that are taken
+              in full or half days (Though a half day leave in the interval
+              will simply be refused - there are no multi-day spanning half-day
+              leaves)
+
+        The method creates one or two new leaves per leave that needs to be
+        split and refuses the original leave.
+        """
+        # Keep track of the original states before refusing the leaves and creating new ones
+        original_states = {l.id: l.state for l in self}
+
+        # Refuse all original leaves
+        self.action_refuse()
+        split_leaves_vals = []
+
+        # Only leaves that span a period outside of the split interval need
+        # to be split.
+        multi_day_leaves = self.filtered(lambda l: l.request_date_from < date_from or l.request_date_to >= date_to)
+
+        for leave in multi_day_leaves:
+            # Leaves in days
+            new_leave_vals = []
+
+            # Get the values to create the leave before the split
+            if leave.request_date_from < date_from:
+                new_leave_vals.append(leave.copy_data({
+                    'request_date_from': leave.request_date_from,
```

Comment:
good point

Reply:
Actually it is necessary after all, because copy_data is overridden on leaves to do 
```python
    def copy_data(self, default=None):
        if default and 'request_date_from' in default and 'request_date_to' in default:
            return super().copy_data(default)
        raise UserError(_('A time off cannot be duplicated.'))
```
So if both aren't past as arguments, you get an error.